packages/flutter_test/lib/src/goldens.dart

```diff
@@ -120,8 +120,61 @@ abstract class GoldenFileComparator {
 ///
 /// Callers may choose to override the default comparator by setting this to a
 /// custom comparator during test set-up (or using directory-level test
-/// configuration). For example, some projects may wish to install a comparator
-/// with tolerance levels for allowable differences.
+/// configuration).
+///
+/// {@tool snippet}
+/// For example, some projects may wish to install a comparator with tolerance
+/// levels for allowable differences:
+///
+/// ```dart
+/// testWidgets('matches golden file with a 0.01 tolerance', (WidgetTester tester) async {
+///   final previousGoldenFileComparator = goldenFileComparator;
+///   goldenFileComparator = _TolerantGoldenFileComparator(
+///     Uri.parse('test/my_widget_test.dart'),
+///     precisionTolerance: 0.01,
+///   );
+///   addTearDown(() => goldenFileComparator = previousGoldenFileComparator);
+///
+///   await tester.pumpWidget(const ColoredBox(color: Color(0xff00ff00)));
+///
+///   await expectLater(
+///     find.byType(ColoredBox),
+///     matchesGoldenFile('my_golden.png'),
+///   );
+/// });
+///
+/// class _TolerantGoldenFileComparator extends LocalFileComparator {
+///   _TolerantGoldenFileComparator(
+///     super.testFile, {
+///     required double precisionTolerance,
+///   }) : _precisionTolerance = precisionTolerance;
```

Comment:
```suggestion
///   }) : assert(precisionTolerance <= 1 || precisionTolerance  0),
            _precisionTolerance = precisionTolerance;
```

Based on the dos below. :)